analytics/client.go

```diff
@@ -0,0 +1,150 @@
+package analytics
+
+import (
+	"github.com/dukex/mixpanel"
+	"github.com/golang/glog"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/segmentio/analytics-go/v3"
+)
+
+type Client interface {
+	// Sends the given tracking event to Segment and Mixpanel (if enabled).
+	TrackEvent(distinctID string, event *Event) error
+
+	// A shorthand wrapper method for TrackEvent that sends a tracking event with the given name and properties.
+	Track(distinctID string, name string, properties map[string]any) error
+}
+
+type clientImpl struct {
+	config        Config
+	segmentClient analytics.Client
+
+	// TODO: Remove Mixpanel once we've confirmed that Segment is working.
+	mixpanelClient mixpanel.Mixpanel
+}
+
+func NewClient(config Config) (Client, error) {
+	analyticsConfig := analytics.Config{
+		DefaultContext: &analytics.Context{
+			App: config.AppInfo,
+		},
+		Endpoint: provideSegmentEndpoint(config.SegmentEndpoint),
+		Logger:   provideLogger(config.IsLoggingEnabled),
+	}
+
+	if config.WriteKey == "" {
+		return nil, errors.New("unable to construct new analytics client. write key cannot be empty")
+	}
+
+	segmentClient, err := analytics.NewWithConfig(config.WriteKey, analyticsConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create segment client")
+	}
+
+	mixpanelClient, err := newMixpanelClient(config)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create mixpanel client")
+	}
+
+	return &clientImpl{
+		config:         config,
+		segmentClient:  segmentClient,
+		mixpanelClient: mixpanelClient,
+	}, nil
+}
+
+func (c clientImpl) TrackEvent(distinctID string, event *Event) error {
```

Comment:
Please document: what is `distinctID`? Looks like this identifies the user. (Why isn't it part of `Event`?)

Reply:
This is the unique identifier for the user. I've chosen the name `distinctID` because I wanted to avoid getting it confused with our `akid.UserID`. I think it makes sense to add this to the event struct. I've updated this in [bfad548](https://github.com/akitasoftware/akita-libs/pull/152/commits/bfad548495f7e2338be10d3e5786fe25df3e4c42)